A79-non-per-call-metrics-architecture.md

```diff
@@ -0,0 +1,432 @@
+    we would want to record metrics for other components as well. Additionally,
+    third-party plugins to gRPC (custom load balancing policies/resolvers) might
+    want to use the same architecture to record metrics as well. (Note that
+    supporting metrics for third-party plugins through this architecture is not
+    an immediate requirement and proposed APIs would start off as
+    internal/experimental, but we might want to make the API public in the
+    future.)
+*   Channel-scoped stats plugins - Applications should be able to register stats
+    plugins such that metrics for only certain channels are recorded on those
+    plugins. There can be multiple stats plugins registered as well.
+*   gRPC/Third-party plugins control the metric instrument definition - As gRPC
+    library/component writers, we want to be in control of the metric definition
+    that gRPC provides, irrespective of the stats plugin used. OpenTelemetry is
+    the only supported stats plugin as of now, but that might change in the
+    future. Third-party plugins to gRPC that choose to use this architecture
+    would be responsible for the definitions of metrics exported by their
+    plugins.
+*   Stats plugins should be able to enable/disable recorded metrics - gRPC has
+    many components that would want to record metrics. Not all of those metrics
+    will be interesting to stats plugins. Stats plugin writers should allow
+    users to configure which metrics are enabled/disabled.
+
+### Proposed Architecture
+
+#### Global Instruments Registry
+
+A new `GlobalInstrumentsRegistry` is added. Any metric instrument that needs to
+be recorded has to register its descriptor to this registry.
+
+Instrument Descriptor -
+
+*   Name (string) - The name of the instrument
+*   Description (string) - The description of the instrument
+*   Unit (string) - The unit for the recordings, e.g. `s` (seconds), `By`
+    (bytes). Follow
+    [OpenTelemetry conventions](https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-units)
+    for consistency.
+*   Type - Implementation defined instrument type, e.g. Long Counter, Double
+    Histogram.
+*   Set of Label Keys (set of strings) - These determine the set of labels that
+    will be present with each recording/measurement.
+*   Set of Optional Label Keys (set of strings) - This is an additional set of
+    labels that stats plugins can choose to either record or throw away based on
+    user configuration. This is useful in cases where certain labels might be
+    deemed not useful for the majority of the cases, and hence shouldn't be
+    recorded by default. (OpenTelemetry allows users to create
+    [custom views](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view)
+    that can filter labels, but in the absence of custom views, the view takes
+    the entire set of labels provided by the instrumentation library.)
+
+On registering an instrument a `GlobalInstrumentHandle` is returned. The
```

Comment:
Nit: please add a comma for clarity: "On registering an instrument, a `...`"

Reply:
Done